OSSMDOC-205: Document configuring federating meshes. (#35580)
openshift/openshift-docs: modules/ossm-federation-overview.adoc

Implementing a federated mesh lets you run, manage, and observe a single service mesh running across multiple OpenShift clusters. {ProductName} federation takes an opinionated approach to a multi-cluster implementation of Service Mesh that assumes _minimal_ trust between meshes.

Service Mesh federation assumes that each mesh is managed individually and retains its own administrator. The default behavior is that no communication is permitted and no information is shared between meshes. The sharing of information between meshes is on an explicit opt-in basis. Nothing is shared in a federated mesh unless it has been configured for sharing. Support functions such as certificate generation, metrics and trace collection remain local in their respective meshes.

You configure the `ServiceMeshControlPlane` on each service mesh to create ingress and egress gateways specifically for the federation, and to specify the trust domain for the mesh.

Federation also involves the creation of additional federation files. The following resources are used to configure the federation between two or more meshes.

* A *ServiceMeshPeer* resource declares the federation between a pair of service meshes.

> * A *ExportedServiceSet* resource declares that one or more services from the mesh are available for use by other meshes.

Reviewer: ```suggestion
* An *ExportedServiceSet* resource declares that one or more services from the mesh are available for use by a peer mesh.
```

> * A *ImportedServiceSet* resource declares which services exported by another mesh will be imported into the mesh.

Reviewer: ```suggestion
* An *ImportedServiceSet* resource declares which services exported by a peer mesh will be imported into the mesh.
```
